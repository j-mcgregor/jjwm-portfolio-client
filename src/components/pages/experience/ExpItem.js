import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import S from '../../styledComponents';

const ExpItem = ({
  company,
  companyUrl,
  city,
  role,
  from,
  to,
  description
}) => (
  <Fragment>
    <S.ExpItem>
      <h5>{role}</h5>
      <h6>
        {companyUrl ? (
          <a href={companyUrl} target="_blank" rel="noopener noreferrer">
            {company}
          </a>
        ) : (
          `${company}`
        )}
      </h6>
      <p>{city}</p>
      <h6 id="date">
        {from} - {to ? `${to}` : 'present'}
      </h6>
      <p>{description}</p>
    </S.ExpItem>
    <hr />
  </Fragment>
);

ExpItem.propTypes = {
  company: PropTypes.string,
  companyUrl: PropTypes.string,
  city: PropTypes.string,
  role: PropTypes.string,
  from: PropTypes.string,
  to: PropTypes.string,
  description: PropTypes.string
};

ExpItem.defaultProps = {
  company: '',
  companyUrl: '',
  city: '',
  role: '',
  from: '',
  to: '',
  description: ''
};

export default ExpItem;
