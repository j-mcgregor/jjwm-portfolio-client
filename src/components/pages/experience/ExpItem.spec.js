import React from 'react';
import { shallow } from 'enzyme';
import ExpItem from './ExpItem';

describe('ExpItem', () => {
  let data = {};
  beforeEach(() => {
    data = {
      company: 'Test Inc',
      companyUrl: 'http://testinc.com',
      city: 'Testland',
      role: 'Tester',
      from: 'Jan 1900',
      to: '',
      description: 'Lorem ipsum'
    };
  });

  it('should render with link if companyUrl present', () => {
    data.to = 'Jan 2000';
    const wrapper = shallow(<ExpItem {...data} />);
    const link = wrapper.find('a[href="http://testinc.com"]');

    expect(wrapper).toMatchSnapshot(wrapper);
    expect(link).toHaveLength(1);
  });

  it('should render with "present" if no "to" date', () => {
    const wrapper = shallow(<ExpItem {...data} />);

    expect(wrapper.find('#date').text()).toBe('Jan 1900 - present');
  });
});
