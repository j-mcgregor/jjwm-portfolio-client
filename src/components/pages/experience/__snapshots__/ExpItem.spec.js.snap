// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpItem should render with link if companyUrl present 1`] = `Object {}`;
